src/test/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/integration/UserMappingServiceTest.kt

```diff
@@ -0,0 +1,167 @@
+    @CsvSource(
+      "N43, N43CP1, NE",
+      "N43, N43CP1, NE",
+      "N43, N43CP2, NE",
+      "N43, N43CPP, NE",
+      "N43, N43LKS, LON",
+      "N43, N43MID, Mids",
+      "N43, N43NTH, NW",
+      "N43, N43SCE, SEE",
+      "N43, N43WSW, SWSC",
+      "N43, N43OMD, Mids",
+      "N43, N43OLK, LON",
+      "N43, N43ONM, NW",
+      "N43, N43OSE, SEE",
+      "N43, N43OWS, SWSC",
+      "N43, N41CRU, NE",
+      "N41, N43CP1, NE",
+      "N41, N43CP1, NE",
+      "N41, N43CP2, NE",
+      "N41, N43CPP, NE",
+      "N41, N43LKS, LON",
+      "N41, N43MID, Mids",
+      "N41, N43NTH, NW",
+      "N41, N43SCE, SEE",
+      "N41, N43WSW, SWSC",
+      "N41, N43OMD, Mids",
+      "N41, N43OLK, LON",
+      "N41, N43ONM, NW",
+      "N41, N43OSE, SEE",
+      "N41, N43OWS, SWSC",
+      "N41, N41CRU, NE",
+      "XX, N43CP1, NE",
+      "XX, N43CP1, NE",
+      "XX, N43CP2, NE",
+      "XX, N43CPP, NE",
+      "XX, N43LKS, LON",
+      "XX, N43MID, Mids",
+      "XX, N43NTH, NW",
+      "XX, N43SCE, SEE",
+      "XX, N43WSW, SWSC",
+      "XX, N43OMD, Mids",
+      "XX, N43OLK, LON",
+      "XX, N43ONM, NW",
+      "XX, N43OSE, SEE",
+      "XX, N43OWS, SWSC",
+      "XX, N41CRU, NE",
+    )
+    fun `determineApArea when user in national probation area and mapping exists for team`(
+      deliusProbationAreaCode: String,
+      deliusTeamCode: String,
+      expectedApArea: String,
+    ) {
+      val staffUserDetails = StaffUserDetailsFactory()
+        .withProbationAreaCode(deliusProbationAreaCode)
+        .withTeams(
+          listOf(
+            StaffUserTeamMembershipFactory().withCode("CODE_NOT_IN_MAPPING").produce(),
+            StaffUserTeamMembershipFactory().withCode(deliusTeamCode).produce(),
+            StaffUserTeamMembershipFactory().withCode("OTHER_CODE_NOT_IN_MAPPING").produce(),
+          ),
+        )
+        .produce()
+
+      val retrievedApArea = ApAreaEntityFactory().produce()
+      every { apAreaRepository.findByIdentifier(expectedApArea) } returns retrievedApArea
+
+      val result = service.determineApArea(
+        usersProbationRegion,
+        staffUserDetails,
+      )
+
+      assertThat(result).isEqualTo(retrievedApArea)
+    }
+
+    @Test
+    fun `determineApArea when user in national probation area but they are not in any teams`() {
+      val staffUserDetails = StaffUserDetailsFactory()
+        .withUsername("J_ALUCARD")
+        .withProbationAreaCode("N43")
+        .withTeams(emptyList())
+        .produce()
+
+      assertThatThrownBy {
+        service.determineApArea(
+          usersProbationRegion,
+          staffUserDetails,
+        )
+      }.hasMessage(
+        "Internal Server Error: Could not find a delius team mapping for delius user J_ALUCARD " +
+          "with delius probation area code N43 and teams []",
+      )
+    }
+
+    @Test
+    fun `determineApArea when user in national probation area but mapping is not available for any of their teams`() {
```

Comment:
would something like:

> throws error and logs details when user in national probation area but mapping is not available for any of their teams

be clearer when scanning the test file?

Reply:
Thanks for the excellent naming suggestions, it's much easier to understand now